File: data/data_creator.py
import os
import numpy as np

# Зададим пределы температуры и давления
min_T, max_T = 1., 1.1
min_p, max_p = 0.0001, 1 # Логарифмический масштаб

# Зададим шаг разбивки по температуре и по давлению
step_T = 0.1
step_p = 10 # Логарифмический масштаб

def range_log(start = 10**-5,finish = 100,step=10):
    """
    Generator object
    Works as range, but instead of add uses multiplies
    """
    now = start
    while now < finish:
        yield now
        now *= step


def main():    
    for p in range_log(min_p,max_p,step_p):
        os.system('lmp_serial < in.lennardjonescrystal -var pressure {:.5f}'.format(p))
        for T in np.arange(min_T,max_T,step_T):
                os.system('lmp_serial < in.restart -var temperature {:.5f} -var pressure {:.5f}'.format(T,p))

if __name__ == '__main__':
    main()

File: data/dataCreator.cpp
//#include "stdafx.h"
#include "fstream"
#include "iostream"
#include "thread" // <-- Чтобы распараллелить
#include "string"
#include <cstdlib>

const double min_T = 0.4;
const double max_T = 1.1;

const double min_p = 0.0001;
const double max_p = 0.11;

const double step_T = 0.1;
const double step_p = 0.0002;


int create(double p){
    std::string command = "lmp_serial < in.lennardjonescrystal -var pressure " + std::to_string(p);
    std::system(command.c_str());
    for (double T = min_T; T < max_T; T = T + step_T)
    {
        command = "lmp_serial < in.restart -var temperature " + std::to_string(T) + " -var pressure " + std::to_string(p);
        std::system(command.c_str());
        //std::cout << command << std::endl;
    }
    return 0;
}

int main(){
    std::ios_base::sync_with_stdio(false);
    for (double p = min_p; p < max_p; p = p + 4*step_p) // так как у меня 4 процессора
    {
        std::thread P1(create, p);
        std::thread P2(create, p + step_p);
        std::thread P3(create, p + 2*step_p);
        std::thread P4(create, p + 3*step_p);
        
        P1.join();
	    P2.join();
	    P3.join();
	    P4.join();
        //std::cout << p << std::endl;
    }

    return 0;
}